File: Dormitories/ClientApp/app/student/Student.ts
import { User } from '../user/User';
import { Faculty } from '../faculty/Faculty';
import { Group } from '../group/Group';
import { Room } from '../room/Room';
import { StudentCategory } from '../studentCategory/StudentCategory';

export class Student {
    id: number;
    fullName: string;
    faculty: Faculty;
    email: string;
    phoneNumber: string;
    studentCardId: string;
    group: Group;
    room: Room;
    studyYear: number;
    studentCategory: StudentCategory;
    user: User;
}

File: Dormitories/Models/Student.cs
namespace Dormitories.Models
{
    public class Student
    {
        public int Id { get; set; }
        public string FullName { get; set; }
        public Faculty Faculty { get; set; }
        public string Email { get; set; }
        public string PhoneNumber { get; set; }
        public string StudentCardId { get; set; }
        public Group Group { get; set; }
        public Room Room { get; set; }
        public int StudyYear { get; set; }
        public StudentCategory StudentCategory { get; set; }
        public User User { get; set; }
    }
}